### PR TITLE
openblas: a small unit test to make sure Lapack symbols are compiled

### PR DESCRIPTION
**Pull request by davydden**
- [ ] move `c` code into a separate file, cleanup a bit.

### CODE REVIEW

**Comment by davydden** [comment]
i realised that i don't know how to get the location of the home directory of the pacakge, i.e. `/path/to/spack/var/spack/repos/builtin/packages/openblas`?

File: var/spack/repos/builtin/packages/openblas/test_cblas_dgemm.c [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#include <cblas.h>
```
Comment (eschnett) [inline var/spack/repos/builtin/packages/openblas/test_cblas_dgemm.c]: Having the C code in its own file is a nice idea!

File: var/spack/repos/builtin/packages/openblas/test_cblas_dgemm.c [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+#include <stdio.h>
+int main(void) {
+int i=0;
+double A[6] = {1.0, 2.0, 1.0, -3.0, 4.0, -1.0};
+double B[6] = {1.0, 2.0, 1.0, -3.0, 4.0, -1.0};
+double C[9] = {.5, .5, .5, .5, .5, .5, .5, .5, .5};
+cblas_dgemm(CblasColMajor, CblasNoTrans, CblasTrans,
+            3, 3, 2, 1, A, 3, B, 3, 2, C, 3);
+for (i = 0; i < 8; i++)
+  printf("%lf ", C[i]);
```
Comment (eschnett) [inline var/spack/repos/builtin/packages/openblas/test_cblas_dgemm.c]: It's common to just write `%f` to output a double-precision value. (`scanf` and `printf` are not symmetric this way.)

File: var/spack/repos/builtin/packages/openblas/test_cblas_dgemm.output [hunk 3]
```diff
@@ -0,0 +1 @@
+11.000000 -9.000000 5.000000 -9.000000 21.000000 -1.000000 5.000000 -1.000000 3.000000
```
Comment (eschnett) [inline var/spack/repos/builtin/packages/openblas/test_cblas_dgemm.output]: Is there a newline missing at the end of the file?

File: var/spack/repos/builtin/packages/openblas/test_cblas_dgemm.c [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+printf("%lf", C[8]);
```
Comment (eschnett) [inline var/spack/repos/builtin/packages/openblas/test_cblas_dgemm.c]: There should be a newline (`\n`) at the end.

I would add newlines after every number; this makes things easier to compare for humans.

File: var/spack/repos/builtin/packages/openblas/test_cblas_dgemm.output (hunk 3 above)
Comment (davydden) [inline var/spack/repos/builtin/packages/openblas/test_cblas_dgemm.output]: no, that's the output of the test, without a new line. But I can of course output a new line in the test as well.

File: var/spack/repos/builtin/packages/openblas/test_cblas_dgemm.c (hunk 1 above)
Comment (davydden) [inline var/spack/repos/builtin/packages/openblas/test_cblas_dgemm.c]: that @alalazo suggestion :smile:

Comment (tgamblin) [inline var/spack/repos/builtin/packages/openblas/test_cblas_dgemm.c]: I like this suggestion...

**Comment by davydden** [comment]
@tgamblin that was WIP :smile: there was a TODO on top and I was also asking a question

**Comment by davydden** [comment]
@tgamblin you don't need to un-merge it, i will fix the inline `c` file once I know how to get path to `/path/to/spack/var/spack/repos/builtin/packages/openblas` from within `package.py`.

**Comment by davydden** [comment]
@alalazo : do you know how to get `/path/to/spack/var/spack/repos/builtin/packages/openblas` from the `package.py` so that i could fix the prematurely merged PR?

**Comment by alalazo** [comment]
@davydden Inspecting the `spack/__init__.py` file I would try to use something like: 
```python
import spack

r = spack.repo  # This should give the path up to /path/to/spack/var/spack/repos/builtin
fullname = join_path(r, spack.repository.packages_dir_name, self.spec.name)
...
```

**Comment by tgamblin** [comment]
@alalazo: don't do that -- it doesn't work from an external repo.  If you are *in* the package, you want this:

```python
os.path.dirname(self.module.__file__)
```

We could probably add a shortcut in `Package` for that.

**Comment by tgamblin** [comment]
@davydden: sorry for merging -- can you put "WIP" in the PR name?

**Comment by davydden** [comment]
@tgamblin NP, will do that next time to make it more visible.

**Comment by alalazo** [comment]
@tgamblin @davydden I removed the comment above, so that people will not accidentaly take the first piece of code found without reading further in the discussion... :smile:

**Pull request merged**